spec/amber/validations/params_spec.cr

```diff
@@ -187,17 +187,25 @@ module Amber::Validators
       end
 
       it "returns validated params on successful validation" do
-        http_params = HTTP::Params.parse("name=elias&last_name=perez&middle=j")
+        http_params = params_builder("name=elias&last_name=perez&middle=j")
         validator = Validators::Params.new(http_params)
         result : Hash(String, String) = {"name" => "elias", "last_name" => "perez"}
 
         validator.validation do
-          required("name") { |v| v.str? & !v.empty? }
-          required("last_name") { |v| v.str? & !v.empty? }
+          required("name") { |v| !v.as(String).empty? }
+          required("last_name") { |v| !v.as(String).empty? }
         end
 
         validator.validate!.should eq result
       end
     end
   end
 end
+
+def params_builder(body = "")
+  params = {} of String | Symbol => Amber::Router::ParamsType
+  HTTP::Params.parse(body).each do |key, value|
+    params[key] = value.as(Amber::Router::ParamsType)
+  end
+  params
```

Comment:
[Enumerable.each_with_object](https://crystal-lang.org/api/0.23.1/Enumerable.html#each_with_object%28obj%2C%26block%29-instance-method) returns that object, so you don't need that extra line to return it.

```crystal
params = {} of String | Symbol => Amber::Router::ParamsType

HTTP::Params.parse(body).each_with_object(params) do |key, value|
  params[key] = value.as(Amber::Router::ParamsType)
end
```

Reply:
I tried this and got an error laving it as is for now

Reply:
Sorry, this is the correct one:

```crystal
params = {} of String | Symbol => Amber::Router::ParamsType

HTTP::Params.parse(body).each_with_object(params) do |(key, value), params|
  params[key] = value.as(Amber::Router::ParamsType)
end

```